### PR TITLE
Database and api pa

### PR DESCRIPTION
**Pull request by panosatha**
- All the pydantic classes are now included in the interface files (solves issues with cylic imports and shows everythings in one place)
- Initial database controller with general methods and measures specific methods.
- Initial API methods to be used in the GUI

### CODE REVIEW

File: flood_adapt/object_model/projection.py [hunk 1]
```diff
@@ -53,6 +43,6 @@ def load_dict(data: dict[str, Any]):
         return obj
 
     def save(self, filepath: Union[str, os.PathLike]):
-        """save Projection to a toml file"""
+        """save Elavate to a toml file"""
```
Comment (GundulaW) [inline flood_adapt/object_model/projection.py:46]:
```suggestion
        """save projection to a toml file"""
```

File: tests/test_database.py
Comment (GundulaW) [inline tests/test_database.py:1]: Are you adding some assert lines to the test?

**Review by GundulaW — APPROVED** [review]
All good, as we had discussed this beforehand already, so I approve already. 

Only a few minor things:
I replaced some "pass" with "raise NotImplementedError". I added these all as suggestions on GitHub so you can easily accept.

Are you adding some assert statements in the database test?

File: tests/test_database.py
Comment (panosatha) [reply tests/test_database.py:1]: added a small test. Most of the tests are in the API

**Pull request merged**